fuel-merkle/src/binary/verify.rs

```diff
@@ -0,0 +1,165 @@
+use crate::binary::node_sum;
+use crate::common::Bytes32;
+use crate::common::ProofSet;
+
+pub fn verify(root: &Bytes32, proof_set: &ProofSet, proof_index: u64, num_leaves: u64) -> bool {
```

Comment:
@bvrooman Do you have ideas on how we can specify the data of the leaf?=)

Reply:
It seems the first item in the set is a "data"
<img width="436" alt="image" src="https://user-images.githubusercontent.com/18346821/226772653-cf2a74aa-da53-45a9-a0bf-e31d63ad7e27.png">

Reply:
With this algorithm, it expects that the first item in the proof set is the leaf that you are proving. 
In your above screenshot, we get item 0 from `proof_set` using the `height` (since `height = 0`), and continue to accumulate the hashes from there.